railties/lib/rails/generators/named_base.rb

```diff
@@ -95,11 +95,11 @@ def regular_class_path # :doc:
         end
 
         def namespaced_class_path # :doc:
-          @namespaced_class_path ||= [namespaced_path] + @class_path
+          @namespaced_class_path ||= namespaced_path + @class_path
         end
 
         def namespaced_path # :doc:
-          @namespaced_path ||= namespace.name.split("::").first.underscore
+          @namespaced_path ||= namespace.name.split("::").map(&:underscore)
```

Comment:
Sorry. I had to revert because this method is public API and now it changes the return from a string to an array. Can you open a new PR keeping this method returning a string?